python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -0,0 +1,372 @@
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    multipleCellCoadd = Output(
+        doc="Output multiple cell coadd",
+        name="{inputWarpName}Coadd{outputCoaddSuffix}",
+        storageClass="MultipleCellCoadd",
+        dimensions=("tract", "patch", "band", "skymap"),
+    )
+
+
+class AssembleCellCoaddConfig(PipelineTaskConfig, pipelineConnections=AssembleCellCoaddConnections):
+    doInterp = Field[bool](doc="Interpolate over NaN pixels?", default=False)
+    interpImage = ConfigurableField(
+        target=InterpImageTask,
+        doc="Task to interpolate images (and extrapolate) over NaN pixels",
+    )
+    scaleZeroPoint = ConfigurableField(
+        target=ScaleZeroPointTask,
+        doc="Task to scale images to a common zero point",
+    )
+    badMaskPlanes = ListField[str](
+        doc="Mask planes that, if set, the corresponding pixel would not be included in the coadd.",
+        default=("BAD", "EDGE", "NO_DATA", "SAT"),
+    )
+    calcErrorFromInputVariance = Field[bool](
+        doc="Calculate coadd variance from input variance by stacking "
+        "statistic. Passed to "
+        "StatisticsControl.setCalcErrorFromInputVariance()",
+        default=True,
+    )
+    max_maskfrac = RangeField[float](
+        doc="Maximum fraction of masked pixels in a cell",
+        default=0.99,
+        min=0.0,
+        max=1.0,
+        inclusiveMin=True,
+        inclusiveMax=False,
+    )
+    # The following config options are specific to the CoaddPsf.
+    coaddPsf = ConfigField(
+        doc="Configuration for CoaddPsf",
+        dtype=CoaddPsfConfig,
+    )
+    inputRecorder = ConfigurableField(
+        doc="Subtask that helps fill CoaddInputs catalogs added to the final Exposure",
+        target=CoaddInputRecorderTask,
+    )
+
+
+class AssembleCellCoaddTask(PipelineTask):
+    ConfigClass = AssembleCellCoaddConfig
+    _DefaultName = "assembleCellCoadd"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.makeSubtask("inputRecorder")
+        self.makeSubtask("interpImage")
+        self.makeSubtask("scaleZeroPoint")
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputData = butlerQC.get(inputRefs)
+
+        if not inputData["inputWarps"]:
+            raise NoWorkFound("No input warps provided for co-addition")
+        self.log.info("Found %d input warps", len(inputData["inputWarps"]))
+        # Construct skyInfo expected by run
+        # Do not remove skyMap from inputData in case _makeSupplementaryData
+        # needs it
+        skyMap = inputData["skyMap"]
+        outputDataId = butlerQC.quantum.dataId
+
+        inputData["skyInfo"] = makeSkyInfo(
+            skyMap, tractId=outputDataId["tract"], patchId=outputDataId["patch"]
+        )
+
+        self.common = CommonComponents(
+            units=CoaddUnits.nJy,
```

Comment:
I don't see anything in the `run` logic that scales the unit to nJy; I think the photometric units are just some arbitrary constant magzero (that I forget) like our other coadds.  An option to scale to nJy in `ScaleZeroPoint` task would be a nice way to address this, as I could see us switching our main coadds to nJy pretty soon if that were available.

Reply:
We will create a `legacy` option that refers to whatever it is currently.